lib/js/components/Pagination/Pagination.vue

```diff
@@ -0,0 +1,396 @@
+			ICON_BUTTON_SIZES: Object.freeze(ICON_BUTTON_SIZES),
+			ICON_BUTTON_COLORS: Object.freeze(ICON_BUTTON_COLORS),
+			ICON_BUTTON_STATES: Object.freeze(ICON_BUTTON_STATES),
+			ICONS: Object.freeze(ICONS),
+		};
+	},
+	computed: {
+		lastPage(): number {
+			return Math.ceil(this.itemsTotalAmount / this.itemsPerPage);
+		},
+		currentPageSanitized(): number {
+			const currentPage = Math.ceil(this.currentPage);
+
+			if (this.pageIsSmallerThanFirstPage(currentPage)) {
+				return FIRST_PAGE_NUMBER;
+			}
+
+			if (this.pageIsLargerThanLastPage(currentPage)) {
+				return this.lastPage;
+			}
+
+			return currentPage;
+		},
+		navigationItems(): Array<number | string> {
+			let delta: number;
+			if (this.lastPage <= MAX_NAVIGATION_ITEMS) {
+				// delta === 7: [1 2 3 4 5 6 7]
+				delta = 7;
+			} else {
+				// delta === 2: [1 ... 4 5 6 ... 10]
+				// delta === 4: [1 2 3 4 5 ... 10]
+				delta = this.currentPage > 4 && this.currentPage < this.lastPage - 3 ? 2 : 4;
+			}
+
+			const range = {
+				start: Math.round(this.currentPage - delta / 2),
+				end: Math.round(this.currentPage + delta / 2),
+			};
+
+			if (range.start - 1 === 1 || range.end + 1 === this.lastPage) {
+				range.start += 1;
+				range.end += 1;
+			}
+
+			let navigationItems: Array<number> =
+				this.currentPage > delta
+					? this.getRange(
+							Math.min(range.start, this.lastPage - delta),
+							Math.min(range.end, this.lastPage),
+					  )
+					: this.getRange(1, Math.min(this.lastPage, delta + 1));
+
+			const withDots = (value, pair) =>
+				navigationItems.length + 1 !== this.lastPage ? pair : [value];
+
+			if (navigationItems[0] !== 1) {
+				navigationItems = withDots(1, [1, ELLIPSIS_FILL]).concat(navigationItems);
+			}
+
+			if (navigationItems[navigationItems.length - 1] < this.lastPage) {
+				navigationItems = navigationItems.concat(
+					withDots(this.lastPage, [ELLIPSIS_FILL, this.lastPage]),
+				);
+			}
+
+			return navigationItems;
+		},
+	},
+	watch: {
+		currentPage(newVal) {
+			this.validatePage(newVal);
+		},
+	},
+	mounted() {
+		this.validatePage(this.currentPage);
+	},
+	methods: {
+		getRange(start: number, end: number): Array<number> {
+			return Array(end - start + 1)
+				.fill(null)
+				.map((_v, i) => i + start);
+		},
+		changePage(page, forceChange = false): void {
+			if (this.currentPageSanitized === page && !forceChange) {
+				return;
+			}
+
+			this.$emit('change-page', page);
+		},
+		isPage(item): boolean {
+			return typeof item === 'number';
+		},
+		pageIsSmallerThanFirstPage(page): boolean {
+			return page < FIRST_PAGE_NUMBER;
+		},
+		pageIsLargerThanLastPage(page): boolean {
+			return page > this.lastPage;
+		},
+		validatePage(page): boolean {
+			if (this.pageIsSmallerThanFirstPage(page)) {
+				this.changePage(FIRST_PAGE_NUMBER, true);
+				return false;
+			}
+
+			if (this.pageIsLargerThanLastPage(page)) {
+				this.changePage(this.lastPage, true);
+				return false;
+			}
+
+			return true;
+		},
```

Comment:
This is a pretty confusing pattern. Validators should validate, not sanitize and validate. How about moving these checks to `changePage` and use that method everywhere?